.github/workflows/continuous-integration.yaml

```diff
@@ -15,41 +15,50 @@ jobs:
     runs-on: '${{ matrix.os }}'
     name: 'Run Testsuite'
     steps:
+#      - name: 'Determine latest MacPorts Version'
+#        id: latest-version
+#        run: |
+#          macports_release_url=$(curl https://raw.githubusercontent.com/macports/macports-base/master/config/RELEASE_URL || curl https://trac.macports.org/export/HEAD/macports-base/config/RELEASE_URL || curl https://distfiles.macports.org/MacPorts/RELEASE_URL)
+#          echo "latest_version=$(echo $macports_release_url | awk -F'/v' '{ print $NF }')" >> $GITHUB_OUTPUT
+#      
       - uses: actions/checkout@v4
       - name: 'Run testsuite'
         run: development/testsuite
 
   install-macports:
+    needs: run-testsuite
     strategy:
       matrix:
         os: ['macos-13', 'macos-14']
-        version: ['2.9.3']
         prefix: ['local']
         include:
           - os: 'macos-14'
-            version: '2.9.3'
             prefix: 'package'
+
     runs-on: '${{ matrix.os }}'
     name: 'Install MacPorts'
-    needs: 'run-testsuite'
     steps:
       - uses: actions/checkout@v4
+      - name: 'Get version From parameters file'
```

Comment:
You do not need to do that here. Add the version selection logic to the action itself and add the detail you're interested in (macports_version) to the action output. It will simplify your design.